v4/newrelic/segments.go

```diff
@@ -199,7 +224,58 @@ func (s *DatastoreSegment) End() {}
 func (s *ExternalSegment) AddAttribute(key string, val interface{}) {}
 
 // End finishes the external segment.
-func (s *ExternalSegment) End() {}
+func (s *ExternalSegment) End() {
+	if s == nil {
+		return
+	}
+	if s.StartTime.span == nil {
```

Comment:
I wonder this check and everything below could be moved into `span.end`? So we just call `s.StartTime.end(s.name())` here? This would avoid code quadruplification. I'm not sure though whether that would be considered idiomatic Go.